examples/src/main/scala/org/apache/spark/examples/snappydata/CreateColumnTable.scala

```diff
@@ -0,0 +1,202 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied. See the License for the specific language governing
+ * permissions and limitations under the License. See accompanying
+ * LICENSE file.
+ */
+package org.apache.spark.examples.snappydata
+
+import java.io.PrintWriter
+
+import com.typesafe.config.Config
+import org.apache.log4j.{Level, Logger}
+
+import org.apache.spark.sql.types.{StringType, DecimalType, IntegerType, StructField, StructType}
+import org.apache.spark.sql.{SnappySession, SparkSession, SnappyContext, SnappyJobValid, SnappyJobValidation, SnappySQLJob}
+
+/**
+ * An example that shows how to create column tables in SnappyData
+ * using SQL or APIs.
+ *
+ * <p></p>
+ * This example can be run either in local mode(in which it will spawn a single
+ * node SnappyData system) or can be submitted as a job to an already running
+ * SnappyData cluster.
+ *
+ * To run the example in local mode go to you SnappyData product distribution
+ * directory and type following command on the command prompt
+ * <pre>
+ * bin/run-example snappydata.CreateColumnTable
+ * </pre>
+ *
+ */
+object CreateColumnTable extends SnappySQLJob {
+
+  case class CustomerRow (C_CUSTKEY: Int, C_NAME: String, C_ADDRESS: String,
+                          C_NATIONKEY: Int, C_PHONE: String, C_ACCTBAL: Double,
+                          C_MKTSEGMENT: String, C_COMMENT: String)
+
+  override def runSnappyJob(snc: SnappyContext, jobConfig: Config): Any = {
+    val pw = new PrintWriter("CreateColumnTable.out")
+    createColumnTableUsingAPI(snc, pw)
+    createColumnTableUsingSQL(snc, pw)
+    createColumnTableInferredSchema(snc, pw)
+    pw.close()
+  }
+
+  override def isValidJob(sc: SnappyContext, config: Config): SnappyJobValidation = SnappyJobValid()
+
+  def createColumnTableUsingAPI(snc: SnappyContext, pw: PrintWriter): Unit = {
+    pw.println()
+
+    pw.println("****Create a column table using API****")
+    // create a partitioned row table using SQL
+    pw.println()
+    pw.println("Creating a column table(CUSTOMER) using API")
+
+    snc.dropTable("CUSTOMER", ifExists = true)
+
+    val schema = StructType(Array(StructField("C_CUSTKEY", IntegerType, false),
+      StructField("C_NAME", StringType, false),
+      StructField("C_ADDRESS", StringType, false),
+      StructField("C_NATIONKEY", IntegerType, false),
+      StructField("C_PHONE", StringType, false),
+      StructField("C_ACCTBAL", DecimalType(15, 2), false),
+      StructField("C_MKTSEGMENT", StringType, false),
+      StructField("C_COMMENT", StringType, false)
+    ))
+
+    // props1 map specifies the properties for the table to be created
+    val props1 = Map("PARTITION_BY" -> "C_CUSTKEY", "BUCKETS" -> "11")
+    snc.createTable("CUSTOMER", "column", schema, props1)
+
+    // insert some data in it
+    pw.println()
+    pw.println("Loading data in CUSTOMER table from a text file with delimited columns")
+    val customerData = snc.sparkContext.
```

Comment:
Quick comment - Instead of RDD loading and parsing, this should be based on DataFrameReader and built in CSV support. How to load itself could be a separate program where you showcase different options (very important and useful). snc.read.csv("path") is all you should need here. If the data files aren't CSV, turn them into CSV.